jobsdb/unionQuery.go

```diff
@@ -83,11 +77,24 @@ func (mj *MultiTenantHandleT) GetAllJobs(workspaceCount map[string]int, params G
 	dsList := mj.getDSList(false)
 	outJobs := make([]*JobT, 0)
 
+	workspacePayloadLimitMap := make(map[string]int64)
+	for workspace, count := range workspaceCount {
+		percentage := math.Max(float64(count), 0) / float64(params.JobsLimit)
+		payloadLimit := percentage * float64(params.PayloadSizeLimit)
+		workspacePayloadLimitMap[workspace] = int64(payloadLimit)
```

Comment:
What would happen if the allotted payload size of a customer is always lesser than the size he has for a single job. Does it make sense to add a safety buffer here of say 1 job to be picked always?

Reply:
this is enforced..
guaranteed minimum of 1 job, maximum up to the limit.

Reply:
To be exact, in `unionQuery`, whenever you are using a `PayloadLimit` we always overflow, i.e. we always return 1 more job than this limit allows